### ISSUE TITLE
Cannot provide defaults for destructured params

### ISSUE THREAD

**Issue by Iodine-** [manwaring/lambda-wrapper#14]
I am not sure if this is an 'issue' per say, but just wanted to raise the point as its something I wasn't expecting.

I was attempting to destructure the `query` param from the `api()` event handler.
```
export const handler = api(async ({ query, success, error }) => {
  const {page = 0, count=10} = query;
}
```

This should work when `query` is undefined, as I expected it to be when I did not provide any in the route.  

however; as the parser [sets the value of non-existent parameters](https://github.com/manwaring/lambda-wrapper/blob/master/src/api/parser.ts#L13) to `null`, the destructuring fails (cannot destructure from null value).

I attempted to set a default value for `query` in the handler definition:
```
export const handler = api(async ({ query = {}, success, error }) => {
  const {page = 0, count=10} = query;
}
```

but as the value of `query` is set to `null`, the default is not applied and the value `query=null` is propagated into the handler function.

I wonder if `undefined` instead of `null` may be better to allow a default value to be specified in the handler signature?

I currently have to handle this case as:
```
export const handler = api(async ({ query, success, error }) => {
  const {page = 0, count=10} = query || {};
}
```
which is not a huge deal, just not what I expected just looking at the examples/docs.

**Comment by manwaring** [comment]
I agree - undefined instead of null is a better default for these props, I'll make that update

thanks for the feedback!

**Comment by manwaring** [comment]
@Iodine- this change has been published to npm as 3.1.0 - I tested it out in a hello world project and it worked there, let me know if you have any issues with it (or any other issues with the library!)

**Comment by Iodine-** [comment]
@manwaring pulled latest and is now functioning much more inline with how I expected it too.  good stuff 👍
